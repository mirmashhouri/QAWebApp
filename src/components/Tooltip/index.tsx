import { useState } from "react";
import styles from "./styles.module.scss";

interface Props {
  content: string;
  children: JSX.Element;
}
const Tooltip = ({ content, children }: Props): JSX.Element => {
  let timeout: any;
  const [active, setActive] = useState(false);

  const showTip = () => {
    timeout = setTimeout(() => {
      setActive(true);
    }, 400);
  };

  const hideTip = () => {
    clearInterval(timeout);
    setActive(false);
  };

  return (
    <div
      className={styles["tooltip-wrapper"]}
      // When to show the tooltip
      onMouseEnter={showTip}
      onMouseLeave={hideTip}
    >
      {/* Wrapping */}
      {children}
      {active && (
        <div className={`${styles["tooltip-tip"]} ${styles.top}`}>
          {/* Content */}
          {content}
        </div>
      )}
    </div>
  );
};

export default Tooltip;
