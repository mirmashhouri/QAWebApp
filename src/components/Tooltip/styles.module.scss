/* Custom properties */
:root {
  --tooltip-text-color: #fff;
  --tooltip-background-color: #000;
  --tooltip-margin: 30px;
  --tooltip-arrow-size: 6px;
}

/* Wrapping */
.tooltip-wrapper {
  display: inline-block;
  position: relative;
}

/* Absolute positioning */
.tooltip-tip {
  background: var(--tooltip-background-color);
  border-radius: 4px;
  color: var(--tooltip-text-color);
  font-family: sans-serif;
  font-size: 14px;
  left: 50%;
  line-height: 1;
  padding: 6px;
  position: absolute;
  // transform: translateX(-50%);
  white-space: nowrap;
  z-index: 100;
}

/* CSS border triangles */
.tooltip-tip::before {
  border: solid transparent;
  border-width: var(--tooltip-arrow-size);
  content: " ";
  height: 0;
  left: 10%;
  margin-left: calc(var(--tooltip-arrow-size) * -1);
  pointer-events: none;
  position: absolute;
  width: 0;
}

/* Absolute positioning */
.tooltip-tip.top {
  top: calc(var(--tooltip-margin) * -1);
}

/* CSS border triangles */
.tooltip-tip.top::before {
  border-top-color: var(--tooltip-background-color);
  top: 100%;
}
