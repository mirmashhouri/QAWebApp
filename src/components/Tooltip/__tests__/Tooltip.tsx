import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import Tooltip from "..";

describe("<Tooltip />", () => {
  it("renders", () => {
    const tree = render(
      <MemoryRouter>
        <Tooltip content="test">
          <div>it is childeren</div>
        </Tooltip>
      </MemoryRouter>
    ).container.firstChild;

    expect(tree).toMatchSnapshot();
  });
});
