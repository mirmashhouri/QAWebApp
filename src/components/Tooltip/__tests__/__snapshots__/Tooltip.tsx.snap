// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> renders 1`] = `
<div
  class="tooltip-wrapper"
>
  <div>
    it is childeren
  </div>
</div>
`;
